### PR TITLE
Remove cw20 support

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
-    CwDexError(#[from] CwDexError),
-
     #[error("{0}")]
     Overflow(#[from] OverflowError),
 
     #[error("{0}")]
     AdminError(#[from] AdminError),
 
+    #[error("{0}")]
+    Asset(#[from] AssetError),
```
Comment (magiodev) [inline src/error.rs:19]: This is unused

File: src/error.rs [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
     Std(#[from] StdError),
 
-    #[error("{0}")]
-    CwDexError(#[from] CwDexError),
-
     #[error("{0}")]
     Overflow(#[from] OverflowError),
 
     #[error("{0}")]
     AdminError(#[from] AdminError),
```
Comment (magiodev) [inline src/error.rs:16]: This is unused. We are using an equivalent from cw-controllers

```
#[error("Caller is not admin")]
    NotAdmin {},
```

File: src/error.rs [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
 #[derive(Error, Debug)]
 pub enum ContractError {
     #[error("{0}")]
     Std(#[from] StdError),
 
-    #[error("{0}")]
-    CwDexError(#[from] CwDexError),
-
     #[error("{0}")]
     Overflow(#[from] OverflowError),
```
Comment (magiodev) [inline src/error.rs:13]: Unused

File: src/error.rs [hunk 4]
```diff
@@ -1,25 +1,23 @@
...
 
     #[error("{0}")]
     AdminError(#[from] AdminError),
 
+    #[error("{0}")]
+    Asset(#[from] AssetError),
+
     #[error("Incorrect amount of native token sent. You don't need to pass in offer_amount if using native tokens.")]
     IncorrectNativeAmountSent,
```
Comment (magiodev) [inline src/error.rs:22]: Unused, also the below ones to this line

File: src/msg.rs [hunk 5]
```diff
@@ -1,100 +1,56 @@
...
     /// the operations that will be executed to perform the swap
-    pub operations: crate::operations::SwapOperationsList,
+    pub operations: Vec<SwapAmountInRoute>,
     /// the amount of tokens that are expected to be received after the swap
     pub return_amount: Uint128,
 }
 
 #[cw_serde]
 #[derive(QueryResponses)]
 pub enum QueryMsg {
```
Comment (magiodev) [inline src/msg.rs:33]: I see what you are aiming for here. Looks good to me in this PR, we can finish that query list on PR #8

File: src/state.rs [hunk 6]
```diff
@@ -1,13 +1,20 @@
...
+    pub denom: String,
+}
 
 /// As an MVP we hardcode paths for each tuple of assets (offer, ask).
 /// In a future version we want to find the path that produces the highest
 /// number of ask assets, but this will take some time to implement.
 /// To support multiple paths between the same asset, we add an id field we increment per asset of the same
 /// path
-pub const PATHS: Map<(AssetInfoKey, AssetInfoKey, u64), SwapOperationsList> = Map::new("paths");
+pub const PATHS: Map<(String, String, u64), Vec<SwapAmountInRoute>> = Map::new("paths");
```
Comment (magiodev) [inline src/state.rs:18]: This should be changed as we said on PR 8 then, removing the unnecessary u64 as we are not excluding using it on the BestPathsForPair query which has been disabled.

**Review by magiodev — CHANGES_REQUESTED** [review]

File: src/msg.rs (hunk 5 above)
Comment (lubkoll) [reply src/msg.rs:33]: yes, we might still want to introduce an alias or our own type (as the SwapAmountInRoute is a bit weird). Just kept it simple for now

File: src/error.rs (hunk 4 above)
Comment (lubkoll) [reply src/error.rs:22]: removed in the PR to the quasar repo

File: src/error.rs (hunk 3 above)
Comment (lubkoll) [reply src/error.rs:13]: removed in the PR to the quasar repo

File: src/error.rs (hunk 1 above)
Comment (lubkoll) [reply src/error.rs:19]: it is used in some asset checks in contract.rs. not sure if we really want these, but kept them for now.

File: src/error.rs (hunk 2 above)
Comment (lubkoll) [reply src/error.rs:16]: yes, that was for supporting the AdminError from cw-controllers. Changed it to use the mars-owner instead (same thing, but a bit nicer implementation)

File: src/msg.rs (hunk 5 above)
Comment (lubkoll) [reply src/msg.rs:33]: we might want to replace this with our own type, not sure yet. The SwapAmountInRoute is a bit annoying as it renames the pool id from `pool_id` to `poolID` during serialization/deserialization and parses u64 as string..

File: src/state.rs (hunk 6 above)
Comment (lubkoll) [reply src/state.rs:18]: I changed it to using `Vec<Vec<SwapAmountInRoute>>` in the other PR. I'd still like to keep the option to have multiple paths, as we need that later and you already wrote almost all of the required code. I only need to find out how to simulate the swap in one query, then I can enable the commented out code again

**Pull request closed without merging**